### DISCUSSION TITLE
Possible to update cache of non-keyable entity?

### DISCUSSION THREAD

**Discussion by sheldonbaker** [urql-graphql/urql#2704] [Q&A] [answered]
The following query is executed against my server (note the lack of an `id` for each entity):

```
query($where: profile_bool_exp!) {
  profilesAggregate(where: $where) {
  	__typename
    
    aggregate {
      __typename
      count
      
      max {
        __typename
        createdAt
      }
    }
  }
}
```

resulting in something like:

```
{
  "data": {
    "profilesAggregate": {
      "__typename": "profile_aggregate",
      "aggregate": {
        "__typename": "profile_aggregate_fields",
        "count": 731,
        "max": {
          "__typename": "profile_max_fields",
          "createdAt": "2022-09-21T06:19:16.683166+00:00"
        }
      }
    }
  }
}
```

Is it somehow possible to use `cache.writeFragment` in an updater to update `profilesAggregate.aggregate.count` from `731` to `732`?

It looks like `cache.writeFragment` requires a `__typename` and `id` to identify the entity involved, but these entities are of course not keyable in that way.

Is the entity somehow reachable by reaching through the parent `Query` that it's embedded in?

Edit: `cache.updateQuery` appears like it could work, but would be extremely unwieldy in this specific case of mine.

**Comment by JoviDeCroock** [comment, marked as answer]
If the parent entity is key-able you can use `writeFragment` with that typename and the correspondign id should def be able to do the trick 😅

**Reply by sheldonbaker** [reply]
Thanks for the quick reply. Unfortunately the parent entity (of type `profilesAggregate` in my example) is also not keyable.

In a perfect world I'd be able to identify the fragment in question by somehow reaching through the root `Query` object (roughly, like `Query.profilesAggregate(someArg: here).aggregate...`) - that's not possible currently, correct?

**Reply by sheldonbaker** [reply]
Looks like I'll have to find out how to make `updateQuery` work, since that's essentially what I'm describing in my comment above.

Thank you!
